src/Workspaces/Core/Portable/Log/LogAggregator.cs

```diff
@@ -1,54 +1,19 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Collections;
-using System.Collections.Concurrent;
-using System.Collections.Generic;
-using System.Linq;
 using System.Threading;
 
 namespace Microsoft.CodeAnalysis.Internal.Log
 {
-    /// <summary>
-    /// helper class to aggregate some numeric value log in client side
-    /// </summary>
-    internal class LogAggregator : IEnumerable<KeyValuePair<object, LogAggregator.Counter>>
+    internal class LogAggregator : AbstractLogAggregator<LogAggregator.Counter>
     {
-        private static int s_globalId;
-
-        private readonly ConcurrentDictionary<object, Counter> _map = new ConcurrentDictionary<object, Counter>(concurrencyLevel: 2, capacity: 2);
-
-        public static int GetNextId()
+        protected override Counter CreateCounter()
         {
-            return Interlocked.Increment(ref s_globalId);
+            return new Counter();
         }
 
-        public static StatisticResult GetStatistics(List<int> values)
+        public int GetAverage(string key)
         {
-            if (values.Count == 0)
-            {
-                return default;
-            }
-
-            var max = int.MinValue;
-            var min = int.MaxValue;
-
-            var total = 0;
-            for (var i = 0; i < values.Count; i++)
-            {
-                var current = values[i];
-                max = max < current ? current : max;
-                min = min > current ? current : min;
-
-                total += current;
-            }
-
-            var mean = total / values.Count;
-            var median = values[values.Count / 2];
-
-            var range = max - min;
-            var mode = values.GroupBy(i => i).OrderByDescending(g => g.Count()).FirstOrDefault().Key;
-
-            return new StatisticResult(max, min, median, mean, range, mode, values.Count);
+            return 0;
```

Comment:
I'm curious why this method is needed?  It doesn't appear to be called anywhere and it simply returns 0 instead of returning an average.

Reply:
It was there before, I assumed it was some sort of place holder. I can remove it.

Reply:
Removed